pkg/agent/controller/packetcapture/bpf/parser.go

```diff
@@ -0,0 +1,154 @@
+package bpf
+
+import (
+	"encoding/binary"
+	"strings"
+
+	"golang.org/x/net/bpf"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+)
+
+func GetRawInstruction(insts []bpf.Instruction) ([]bpf.RawInstruction, error) {
+	return bpf.Assemble(insts)
+
+}
+
+// CompilePacketFilter compile our crd spec to bpf instructions. For now, we only focus on
+// ipv4 traffic. Compare to raw BPF filter supported by libpcap, we only need to support
+// limited usercases so an expression parser is not needed.
+func CompilePacketFilter(spec *crdv1alpha1.PacketCaptureSpec, matchPacket *binding.Packet) []bpf.Instruction {
```

Comment:
nit: you only need the packet here, not the full spec

Reply:
updated.